pkg/services/authn/clients/ext_jwt_test.go

```diff
@@ -0,0 +1,514 @@
+			payload: rfc9068Payload{
+				Issuer:   "http://localhost:3000",
+				Subject:  "user:id:2",
+				Audience: jwt.Audience{"http://localhost:3000"},
+				ID:       "1234567890",
+				Scopes:   []string{"profile", "groups"},
+				Expiry:   time.Date(2023, 5, 3, 0, 0, 0, 0, time.UTC).Unix(),
+				IssuedAt: time.Date(2023, 5, 2, 0, 0, 0, 0, time.UTC).Unix(),
+			},
+		},
+		{
+			name: "missing iat",
+			payload: rfc9068Payload{
+				Issuer:   "http://localhost:3000",
+				Subject:  "user:id:2",
+				Audience: jwt.Audience{"http://localhost:3000"},
+				ID:       "1234567890",
+				ClientID: "grafana",
+				Scopes:   []string{"profile", "groups"},
+				Expiry:   time.Date(2023, 5, 3, 0, 0, 0, 0, time.UTC).Unix(),
+			},
+		},
+		{
+			name: "iat later than current time",
+			payload: rfc9068Payload{
+				Issuer:   "http://localhost:3000",
+				Subject:  "user:id:2",
+				Audience: jwt.Audience{"http://localhost:3000"},
+				ID:       "1234567890",
+				ClientID: "grafana",
+				Scopes:   []string{"profile", "groups"},
+				Expiry:   time.Date(2023, 5, 3, 0, 0, 0, 0, time.UTC).Unix(),
+				IssuedAt: time.Date(2023, 5, 2, 0, 2, 0, 0, time.UTC).Unix(),
+			},
+		},
+		{
+			name: "missing jti",
+			payload: rfc9068Payload{
+				Issuer:   "http://localhost:3000",
+				Subject:  "user:id:2",
+				Audience: jwt.Audience{"http://localhost:3000"},
+				ClientID: "grafana",
+				Scopes:   []string{"profile", "groups"},
+				Expiry:   time.Date(2023, 5, 3, 0, 0, 0, 0, time.UTC).Unix(),
+				IssuedAt: time.Date(2023, 5, 2, 0, 0, 0, 0, time.UTC).Unix(),
+			},
+		},
+		{
+			name: "unsupported alg",
+			payload: rfc9068Payload{
+				Issuer:   "http://localhost:3000",
+				Subject:  "user:id:2",
+				Audience: jwt.Audience{"http://localhost:3000"},
+				ID:       "1234567890",
+				ClientID: "grafana",
+				Scopes:   []string{"profile", "groups"},
+				Expiry:   time.Date(2023, 5, 3, 0, 0, 0, 0, time.UTC).Unix(),
+				IssuedAt: time.Date(2023, 5, 2, 0, 0, 0, 0, time.UTC).Unix(),
+			},
+			alg: jose.RS384,
+		},
+	}
+
+	extJwtClient := setupTestCtx(t, nil, nil)
+	mockTimeNow(time.Date(2023, 5, 2, 0, 1, 0, 0, time.UTC))
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.alg == "" {
+				tc.alg = jose.RS256
+			}
+			tokenToTest := generateToken(tc.payload, pk, tc.alg)
+			_, err := extJwtClient.verifyRFC9068Token(context.Background(), tokenToTest)
+			require.Error(t, err)
+		})
+	}
+}
+
+func setupTestCtx(t *testing.T, userSvc user.Service, cfg *setting.Cfg) *ExtendedJWT {
+	if cfg == nil {
+		cfg = &setting.Cfg{
+			ExtendedJWTAuthEnabled:    true,
+			ExtendedJWTExpectIssuer:   "http://localhost:3000",
+			ExtendedJWTExpectAudience: "http://localhost:3000",
+		}
+	}
+
+	signingKeysSvc := &signingkeystest.FakeSigningKeysService{}
+	signingKeysSvc.ExpectedServerPublicKey = &pk.PublicKey
+
+	extJwtClient := ProvideExtendedJWT(userSvc, cfg, signingKeysSvc)
+	return extJwtClient
+}
+
+type rfc9068Payload struct {
```

Comment:
Nit. Think we could use `ExtendedJWTClaims` instead.

Reply:
Done, thanks!